towerlib/entities/job.py

```diff
@@ -1227,20 +1227,19 @@ def launch(self, extra_vars=None, job_tags=None, limit=None, inventory=None, cre
         return Job(self._tower, response.json())
 
     @inventory.setter
-    def inventory(self, value):
+    def inventory(self, name):
         """Inventory applied as a prompt, assuming job template prompts for inventory.
-
+        
         Returns:
             None.
-
+            
         """
-        inventory_id = value
-        if isinstance(value, Inventory):
-            inventory_id = value.id
-
-        self._update_values('inventory', inventory_id)
+        inventory = next(self._tower.get_inventories_by_name(name))  # pylint: disable=protected-access
```

Comment:
This will randomly pick the first inventory returned which would match the name. Inventories can have conflicting names as they are organization based. So this will randomly fail. The call to target an inventory is https://github.com/schubergphilis/towerlib/blob/4ed562dd7e08022cddb5176cb9e67e011aafa33a/towerlib/towerlib.py#L804 which also requires an organization.

Reply:
@ttavi21 will you have a look at the above?

Reply:
@ttavi21 I am trying to get a new version of the library out this week or the next one, will you please have a look at the above?

Reply:
@costastf Hello. Yes, I will take a look in 2-3 days.

Reply:
Lovely! Then hopefully by next week i will have fixed the testing also and we can release a new version.